compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -681,6 +681,15 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             Some(self_ty),
         );
 
+        let polarity = match polarity {
+            rustc_ast::BoundPolarity::Positive => ty::PredicatePolarity::Positive,
+            rustc_ast::BoundPolarity::Negative(_) => ty::PredicatePolarity::Negative,
+            rustc_ast::BoundPolarity::Maybe(_) => {
+                // No-op.
+                return arg_count;
```

Comment:
Doesn't this early return mean we don't validate associated item constraints which get lowered below this point in `lower_assoc_item_constraint`. E.g., is it still the case that `?Sized<Undefined = (), Undefined = ()>` gets accepted under this PR?

Reply:
:'( yeah

Reply:
OK, so I made it so that we at least *validate* these associated type bounds, but not outright deny them. We may want to do that later.